doc/User_Management/User_Management.md

```diff
@@ -0,0 +1,347 @@
+
+This ensures that local user accounts are consistently applied with robust security postures and persist across reboots and upgrades.
+
+## 2. Definitions/Abbreviations
+
+| Abbreviation | Definition                                                                |
+|:------------:|:--------------------------------------------------------------------------|
+| **`userd`** | The new User Daemon proposed in this design.                              |
+| **PAM** | Pluggable Authentication Modules                                          |
+
+
+## 3. Overview
+This feature introduces a new dedicated daemon, **`userd`**, which manages the full lifecycle of local users based on definitions in `CONFIG_DB`. It simplifies management by providing a user-friendly CLI, mapping abstract roles (`administrator`, `operator`) to specific Linux groups, and enforcing security policies. This provides a solution for managing secure, persistent local user accounts.
+
+## 4. Requirements
+### 4.1 Functional Requirements
+1.  The system must allow an administrator to define a local user account declaratively.
+2.  The user definition must support:
+    * Username and a pre-hashed password.
+    * A role, limited to either **`administrator`** or **`operator`**.
+    * Authorized SSH keys (statically defined).
+3.  The system must map roles to underlying Linux groups:
+    * `administrator`: members of `sudo`, `docker`, `redis` and `admin` groups.
+    * `operator`: members of a standard, non-privileged and would belong to `users` group.
+4.  The system will **auto-generate** a unique UID for each new user.
+5.  There should be atleast one user configured in the role of `administrator`
+6.  **Default Admin User:**
+    * The **`admin`** user (or the user specified during compilation) must be included in the `init_cfg.json` with default settings.
+    * This ensures the admin user is present in CONFIG_DB by default and can be managed like any other user.
+    * The admin user can be modified or disabled or deleted through standard CONFIG_DB operations.
+7.  **Security Policy Requirements:**
+    * **Login Attempts:** The system must support configuring a global maximum number of failed login attempts per role before accounts are temporarily locked.
+8.  User accounts and their configurations must persist across system reboots and upgrades.
+9.  **System Consistency:** On startup, the system must perform a consistency check to ensure Linux users match CONFIG_DB definitions and automatically remove any users that were added directly to Linux (bypassing CONFIG_DB).
+
+## 5. Architecture Design
+The architecture centers on the new `userd` daemon. This daemon will now interact with several core Linux subsystems to enforce the configured security policies.
+
+**`userd`'s Points of Interaction:**
+1.  **CONFIG_DB:** Single source of truth for user configuration and global security policies.
```

Comment:
Is this feature controlled by the feature flag? >> right now it is controlled by init.cfg